packages/adapters/src/adapters/Cornerstone3D/Segmentation/generateSegmentation.ts

```diff
@@ -0,0 +1,57 @@
+import { normalizers, derivations } from "dcmjs";
+import { fillSegmentation } from "../../Cornerstone/Segmentation_4X";
+
+const { Normalizer } = normalizers;
+const { Segmentation: SegmentationDerivation } = derivations;
+
+/**
+ * generateSegmentation - Generates a DICOM Segmentation object given cornerstoneTools data.
+ *
+ * @param images - An array of the cornerstone image objects, which includes imageId and metadata
+ * @param labelmaps - An array of the 3D Volumes that contain the segmentation data.
+ */
+function generateSegmentation(images, labelmaps, metadata, options) {
+    const segmentation = _createMultiframeSegmentationFromReferencedImages(
+        images,
+        metadata,
+        options
+    );
+    return fillSegmentation(segmentation, labelmaps, options);
+}
+
+/**
+ * _createMultiframeSegmentationFromReferencedImages - description
+ *
+ * @param images - An array of the cornerstone image objects.
```

Comment:
Could you add a bit more information here on what the images are - I think given the context that the images are the segmentation images, and are thus single bit image objects containing the actual data in the segmentation, but it isn't at all obvious to me externally to this function that the images are just the segmentation single bit images.

Reply:
Looking at the dcmjs side, these are the original images, and in fact the generation code for the derivation throws away all the pixel data itself, using only the length.  I suggest not passing in the actual pixel data, but generating a dummy buffer containing just the byte lengths as required.   That way, even the temporary buffer will have the right lengths, as the generated buffer is simply a copy of the derived dataset buffers, and that is usually twice the required length.

Reply:
we already have the pixelData from the volume. So the cost is negligible compared to creating another temporary buffer etc.